src/librbd/io/ObjectRequest.cc

```diff
@@ -257,9 +280,11 @@ void ObjectReadRequest<I>::read_parent() {
 
   // calculate reverse mapping onto the image
   Extents parent_extents;
-  Striper::extent_to_file(image_ctx->cct, &image_ctx->layout,
-                          this->m_object_no, this->m_object_off,
-                          this->m_object_len, parent_extents);
+  for (auto [object_off, object_len]: this->m_extents) {
+    Striper::extent_to_file(image_ctx->cct, &image_ctx->layout,
```

Comment:
Hmm -- I have to think about this more to see if some odd striping patterns would lead to odd behavior. Right now that function has an expectation that it's given a clean (empty) vector.